### PR TITLE
Polpatrol instrumentation

### PR DESCRIPTION
**Pull request by ritzdorf**
Modifications that expose additional information to the PolPatrol analysis tool.

>   - What does it do?

Adds logging that tracks:

- Calls to environment functions (tagged by category)
- Wasm Memory consumption

>   - What important points reviewers should know?

We think that as these are `info`-level logs they should not negatively influence performance.

Connected to this PR for polkadot: https://github.com/paritytech/polkadot/pull/700

### CODE REVIEW

File: srml/im-online/src/lib.rs [hunk 1]
```diff
@@ -181,10 +181,10 @@ pub type AuthIndex = u32;
 pub struct Heartbeat<BlockNumber>
 	where BlockNumber: PartialEq + Eq + Decode + Encode,
 {
-	block_number: BlockNumber,
-	network_state: OpaqueNetworkState,
-	session_index: SessionIndex,
-	authority_index: AuthIndex,
+	pub block_number: BlockNumber,
```
Comment (rphmeier) [inline srml/im-online/src/lib.rs]: The newly-public fields should come with doc-strings.

File: core/executor/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "substrate-executor"
-version = "2.0.0"
+version = "2.0.1-gw001"
```
Comment (bkchr) [inline core/executor/Cargo.toml]:
```suggestion
version = "2.0.0"
```

File: core/executor/src/wasm_executor.rs [hunk 3]
```diff
@@ -367,12 +374,17 @@ struct SubstrateExternals;
 impl_wasm_host_interface! {
 	impl SubstrateExternals where context {
 		ext_malloc(size: WordSize) -> Pointer<u8> {
+			info!("DEBUG_CS_INSTRUMENTATION {}:{} TAG_Memory ext_malloc", file!(), line!());
```
Comment (bkchr) [inline core/executor/src/wasm_executor.rs]: First, these functions are not used anymore.
Second, using `info!` to print debug information is not good.

Why not use grafana to collect this stuff?

**Comment by bkchr** [comment]
As spoken in riot, they will look into the new tracing infrastructure of substrate to integrate their changes. :)

**Comment by bkchr** [comment]
Will close it for now, feel free to reopen/recreate when you have something new.

**Pull request closed without merging**